corehq/apps/data_analytics/malt_generator.py

```diff
@@ -63,25 +66,28 @@ def _get_malt_row_dicts(self, domain_name, monthspan, users_by_id):
             apps_submitted_for = get_app_submission_breakdown_es(domain_name, monthspan, users)
             for app_row in apps_submitted_for:
                 app_id = app_row.app_id
+                user_id = app_row.user_id
                 num_of_forms = app_row.doc_count
                 try:
                     app_data = self._app_data(domain_name, app_id)
-                    user_id, username, user_type, email = self._user_data(
-                        app_row.user_id,
-                        app_row.username,
-                        users_by_id
-                    )
+                    if app_row.user_id not in users_by_id:
+                        # highly unlikely, but adding a safety check for a recent refactor
+                        notify_exception(None, f"A form submission for user with id {user_id} was returned by "
+                                               f"get_app_submission_breakdown_es but was not found in the "
+                                               f"users_by_id parameter.")
```

Comment:
What visibility do `notify_exception()` events receive? Is there a guaranteed investigation into the exception? If not, I might advocate a hard (uncaught) exception that fails this part of the report so the implications of this condition cannot go unaddressed by the consumers of the report.

Reply:
I'm tempted to just remove that check entirely because it is essentially showing a mistrust in ElasticSearch results/filtering and the same level of awareness can be achieved by just attempting to access `users_by_id[app_row.user_id]`. Granted that would cause the entire task to fail if the id is does not exist, but that would actually provide better visibility in that we would see malt generation celery task failures in datadog. I could even setup an alert in datadog to fire if that task fails. 

If we wanted to stick with something closer to the current implementation, I believe there is a way to send an email for specific exceptions, but again my stance at the moment is that this code is unnecessary to continue maintaining. 

@joxl @mjriley let me know if you feel differently.

Reply:
> Granted that would cause the entire task to fail if the id is does not exist

:heavy_plus_sign: